packages/tx/src/fake.ts

```diff
@@ -1,62 +0,0 @@
-import { toBuffer } from 'ethereumjs-util'
-import { Buffer } from 'buffer'
-import { BufferLike, FakeTxData, PrefixedHexString, TransactionOptions } from './types'
-import Transaction from './transaction'
-
-/**
- * Creates a new transaction object that doesn't need to be signed.
- *
- * @param data - A transaction can be initialized with its rlp representation, an array containing
- * the value of its fields in order, or an object containing them by name.
- *
- * @param opts - The transaction's options, used to indicate the chain and hardfork the
- * transactions belongs to.
- *
- * @see Transaction
- */
-export default class FakeTransaction extends Transaction {
```

Comment:
This `FakeTransaction` class is the only way of implementing `eth_call` using `ethereumjs-vm` now, because of this issue: https://github.com/ethereumjs/ethereumjs-vm/issues/613

This is implemented using `FakeTransaction` in both Ganache and Buidler EVM.

I think it's great dropping this here, but I think we should document how `eth_call` can be implemented. Maybe it can be an example in the VM package.

Reply:
Thanks, that sounds good I like the idea of an adding an example of this.

An implementation of `FakeTransaction` used in `runTx` would look something like this (just writing here):

```javascript
import { Transaction, Address } from '@ethereumjs/tx'

public async _getFakeTransaction(
    txParams: TransactionParams
  ): Promise<Transaction> {
  const from = new Address(txParams.from)
  delete txParams.from

  const tx = Transaction.fromTxData(txParams, this._common)

  const fakeTx = Object.create(tx)

  // override getSenderAddress
  fakeTx.getSenderAddress = function () { return from }

  return fakeTx
}
```